### PR TITLE
FixedSizeStream decorating FixedSizeStream

### LINKED ISSUE #359: FixedSizeStream::create does not respect the chunk size when wrapping a FixedSizeStream
The factory method has an optimization to return the wrapped stream directly if it is already a FixedSizeStream. But that should be done only if the provided FixedSizeStream has the same fixed size than the requested one IMO.

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #359

### CODE REVIEW

File: src/Core/tests/Unit/Stream/FixedSizeStreamTest.php [hunk 1]
```diff
@@ -43,6 +43,16 @@ public function testChunk(RequestStream $content, int $size, array $expected): v
         self::assertSame($expected, \iterator_to_array($stream));
     }
 
+    /**
+     * @dataProvider provideChunks
+     */
+    public function testDecoratingFixedSize(RequestStream $content, int $size, array $expected): void
+    {
+        $stream = FixedSizeStream::create(FixedSizeStream::create($content, \random_int(1, 10)), $size);
```
Comment (Nyholm) [inline src/Core/tests/Unit/Stream/FixedSizeStreamTest.php]: We don’t need the inner stream to be random here. Do we?

We don’t want our tests to fail randomly

Comment (jderusse) [reply src/Core/tests/Unit/Stream/FixedSizeStreamTest.php]: it don't have to be random, but it should work whatever the value of the inner chunk size.
if the test failed randomly, then their is a bug

Comment (stof) [reply src/Core/tests/Unit/Stream/FixedSizeStreamTest.php]: well, we should have 2 non-random tests: one using the same size, and one using always a different size.

Comment (jderusse) [reply src/Core/tests/Unit/Stream/FixedSizeStreamTest.php]: removed.

**Comment by Nyholm** [comment]
Could you rebase and merge please?

**Pull request merged**